corehq/form_processor/models.py

```diff
@@ -54,16 +105,34 @@ def _to_xml_element(payload):
             return etree.fromstring(payload)
         return _to_xml_element(xml)
 
-    @property
-    def form_data(self):
-        from .utils import convert_xform_to_json
-        xml = self._get_xml()
-        return convert_xform_to_json(xml)
-
     def _get_xml(self):
         xform_attachment = self.xformattachmentsql_set.filter(name='form.xml').first()
         return xform_attachment.read_content()
 
+    def archive(self, user=None):
+        if self.is_archived:
+            return
+        with transaction.atomic():
+            self.state = self.ARCHIVED
+            self.xformoperationsql_set.create(
+                user=user,
+                operation=XFormOperationSQL.ARCHIVE,
+            )
+            self.save()
+        xform_archived.send(sender="form_processor", xform=self)
+
+    def unarchive(self, user=None):
+        if not self.is_archived:
+            return
+        with transaction.atomic():
+            self.state = self.NORMAL
+            self.xformoperationsql_set.create(
+                user=user,
+                operation=XFormOperationSQL.UNARCHIVE,
+            )
+            self.save()
+        # xform_unarchived.send(sender="form_processor", xform=self)
+
 
 class XFormAttachmentSQL(models.Model):
     attachment_uuid = models.CharField(max_length=255, unique=True, db_index=True)
```

Comment:
yeah that's the way i originally had it and changed because i was thinking there might be some failure state where the attachment gets written and the form fails to save or something, but i don't think that's possible on second thought. will update